### ISSUE TITLE
Tree renders with Vue errors in the browser's console

### ISSUE THREAD

**Issue by dandee** [grapoza/vue-tree#228]
**Describe the bug**
Tree renders with Vue errors in the browser's console with the following error:

`TypeError: current.treeNodeSpec is undefined`

**To Reproduce**
Install version 3.0.3. Build app. Run.

Code:

```
<tree-view id='tree' :load-nodes-async='loadTree' :model-defaults='treeDefaults' :selectionMode='"single"'>
  <template v-slot:loading-root>
    <div class='d-flex justify-content-center'>
      <div class='spinner-border text-primary' role='status'>
        <span class='sr-only'>Loading</span>
      </div>
    </div>
  </template>
</tree-view>
```

**Expected behavior**
Tree renders without errors.

**Additional context**
Here's the excerpt from my app that reproduces the issue:

https://github.com/dandee/vue-tree-repro

to run it execute:

```
npm install
npm start
```

The error I've reported is being reported by the latest Firefox. Chromium is giving me a different, more confusing error messages.

**Comment by grapoza** [comment]
Yup, that's definitely a bug I'll have to look into this weekend. I should have tried downgrading that to 3.0.3 myself before responding to your earlier bug, so that's on me. Sorry for that. Hopefully tomorrow afternoon I can start getting this sorted out, if I can I'll try to have it done by your Monday morning. Fingers crossed.
